### PR TITLE
Add test pipeline for powershell scripts

### PR DESCRIPTION
**Pull request by sima-zhu**
# Contributing to the Azure SDK
Testing pipeline: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2116919&view=logs&j=17b53491-05ec-5417-97d3-b5dd98f6c69f

All tests with tag `UnitTest` under scripts are covered. 
Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: eng/common/pipelines/templates/steps/run-pester-tests.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ steps:
         $config.CodeCoverage.Enabled = $true
         $config.TestResult.Enabled = $true
         $config.Run.PassThru = $true
+        $config.Output.Verbosity = 'Detailed'
```
Comment (sima-zhu) [inline eng/common/pipelines/templates/steps/run-pester-tests.yml:28]: Testing logging purpose

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/autorest.md [hunk 2]
```diff
@@ -9,7 +9,9 @@ Run `generate.ps1` or `dotnet msbuild /t:GenerateCode` to generate code.
 public-clients: true
 title: DeviceUpdate
 
-input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/d7c9be23749467be1aea18f02ba2f4948a39db6a/specification/deviceupdate/data-plane/Microsoft.DeviceUpdate/stable/2022-10-01/deviceupdate.json
+require:
```
Comment (weshaggard) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/autorest.md:12]: This looks unrelated to your other changes.

**Comment by weshaggard** [comment]
Lets make sure the new scripts piepline is hooked up and tested before merging.

File: eng/scripts/automation/test/GenerateAndBuildLib-functions.tests.ps1 [hunk 3]
```diff
@@ -21,7 +21,7 @@ BeforeAll {
     . $PSScriptRoot/../GenerateAndBuildLib.ps1
 }
 
-Describe "update autorest.md" -Tag "Unit" {
+Describe "update autorest.md" -Tag "UnitTest" {
```
Comment (sima-zhu) [inline eng/scripts/automation/test/GenerateAndBuildLib-functions.tests.ps1:24]: Open an issue to address the disabled tests in this PR.
https://github.com/Azure/azure-sdk-for-net/issues/33487

File: eng/scripts/automation/test/GenerateAndBuildLib-functions.tests.ps1 [hunk 4]
```diff
@@ -105,6 +105,7 @@ Describe "update autorest.md" -Tag "Unit" {
     }
 }
 
+# Open an issue to track the disabled tests here: https://github.com/Azure/azure-sdk-for-net/issues/33487
 Describe "Generate and Build SDK" -Tag "Unit" {
```
Comment (sima-zhu) [inline eng/scripts/automation/test/GenerateAndBuildLib-functions.tests.ps1:109]: Open an issue to address the disabled tests in this PR.
https://github.com/Azure/azure-sdk-for-net/issues/33487

**Review by weshaggard — APPROVED** [review]
